Issue/4413 onboarding generic error

nbradbury opened the pull request:
Parent issue #4413 - this PR implements the generic error state for the onboarding flow. Note that there were no designs for this screen, so a design review is required.

![generic](https://user-images.githubusercontent.com/3903757/128061290-c6c3ea5e-324e-4bfa-8bde-4d4a6b07a9a5.png)

PS: The above screenshot was updated to reflect the layout changes in 9f072b4.


Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

nbradbury committed:
First pass at generic error screen

nbradbury committed:
Added close button

nbradbury committed:
Shortened label text

nbradbury committed:
Added illustration to generic state

nbradbury committed:
Tweaked margins

nbradbury committed:
Suppress the LongMethod warning

nbradbury committed:
Tweaked margins (again)

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingViewModel.kt:
```diff
@@ -81,7 +83,11 @@ class CardReaderOnboardingViewModel @Inject constructor(
+                    )
                 CardReaderOnboardingState.NoConnectionError ->
                     viewState.value = OnboardingViewState.NoConnectionErrorState
             }
```
Not your change, but just to consider in the next PRs - make it `.exhaustive`  maybe?

kidinov commented on WooCommerce/src/main/res/layout/fragment_card_reader_onboarding_stripe.xml:
```diff
@@ -1,81 +1,86 @@
-            app:layout_constraintTop_toTopOf="parent"
-            app:layout_constraintVertical_bias="1" />
+    <include
+        android:id="@+id/learn_more_container"
```
nit: shell we call it `learnMoreContainer` to follow the view's IDs convention?

kidinov reviewed (approved):
Looks good! Just 2 things to consider:

* Shell be the content be centered between the top and "close" button? Maybe I'm wrong but it looks like it's the case for other error screens
* Maybe as part of another PR probably would be nice to check if all View IDs and the layouts (currently they are `fragment...` instead of `..._fragment` for instance) are following the convention we voted a couple of weeks ago - https://github.com/woocommerce/woocommerce-android/blob/develop/CONVENTION.md#view-ids

**Not merging because design review is needed**

nbradbury committed:
Merge branch 'issue/4413-pending-requirements-button' of https://github.com/woocommerce/woocommerce-android into issue/4413-onboarding-generic-error

nbradbury committed:
Updated layout to more closely match other onboarding screens

nbradbury commented:
> Shell be the content be centered between the top and "close" button? Maybe I'm wrong but it looks like it's the case for other error screens

@kidinov You're right, that screen is a little off. I updated it to more closely match the other onboarding screens in 9f072b4. I also made the `when` clause `exhaustive` and renamed the view to `learnMoreContainer`, as you suggested. Ready for another round!

![generic](https://user-images.githubusercontent.com/3903757/128060360-3c7453b0-36f8-45d0-be6b-28cdb3c0a88b.png)

nbradbury committed:
Made when clause exhaustive

nbradbury committed:
Fleshed out network error state

nbradbury committed:
Added retry button

nbradbury committed:
Added landscape button layout

nbradbury committed:
Added importantForAccessibility="no" to the illustration

kidinov commented:
@nbradbury it's approved, from my side, let's see what @woocommerce/mobile-designers will say =)

ghost commented:
Hey @nbradbury @kidinov , I've done a design for Android and iOS for this error. Could we please use this instead.

CC: @Garance91540 

<img width="1382" alt="Screen Shot 2021-08-04 at 4 49 47 pm" src="https://user-images.githubusercontent.com/74938219/128135224-36595615-8679-4dd9-a416-a76d6df68c2c.png">

Garance91540 reviewed (approved):
Agreeing with Adam's changes ;)

Garance91540 reviewed (changes requested):
^

malinajirka committed:
Implement no connection logic in onboarding checker

nbradbury commented:
@adamzelinski @Garance91540 I'll update this to match the design (thank you!), but I'm wondering why we need to say "Unable to verify In-Person Payments for this store" twice?

nbradbury committed:
Removed Close button from generic error screen and updated header

nbradbury committed:
Updated generic screen to match new design

nbradbury commented:
@kidinov I've updated the screen to match the above design. If it's okay with you, I'd prefer to merge this now and if there's any feedback about the wording we can address that later.

![generic](https://user-images.githubusercontent.com/3903757/128178209-a35e2f42-f90a-48a6-aac3-227db6c3d7bc.png)

nbradbury committed:
Removed close button

kidinov commented:
@nbradbury totally ok with me =)

nbradbury committed:
Merge branch 'develop' of https://github.com/woocommerce/woocommerce-android into issue/4413-onboarding-generic-error

Conflicts:
	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt

nbradbury committed:
Resolved conflicts

nbradbury committed:
Merge branch 'issue/4413-onboarding-generic-error' of https://github.com/woocommerce/woocommerce-android into issue/4413-onboarding-network-error

Conflicts:
	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt

nbradbury committed:
Resolved conflicts

nbradbury committed:
Merge pull request #4551 from woocommerce/issue/4413-onboarding-network-error

issue/4413-onboarding-network-error

nbradbury committed:
Merge pull request #4556 from woocommerce/issue/4413-onboarding-network-error-logic

Implement no connection logic in onboarding checker
